src/app/views/sportsman/Components/create-sportsman/create-sportsman.component.ts

```diff
@@ -61,31 +146,56 @@ export class CreateSportsmanComponent implements OnInit {
         toast.addEventListener('mouseleave', Swal.resumeTimer);
       },
     });
-    
-    const formSportsman = {      
+
+    const formSportsman = {
       ...this.sportsmansForm.value,
       image: 'default.png',
-      category: this.sportsmansForm.value.category.name,
-      CategoriumID: this.sportsmansForm.value.category.code
+      category: this.sportsmansForm.value.category,
+      CategoriumID: this.categorias.find(item => item.name == this.sportsmansForm.value.category)?.code,
+      ID: this.dataID
     };
-    this.sporsmanService$.createSportsman(formSportsman).subscribe(
-      async (res) => {
-        await Toast.fire({
-          icon: 'success',
-          title: `${res.Menssage}`,
-        });
-        this.sportsmansForm.reset();
-        this.currentPage = 0;
-        this.CreateSportsman.emit(true);
+    if (!this.isEdit) {
```

Comment:
en toda esta seccion estas repitiendo el mismo codigo pues hacerlo asi this.sporsmanService$[this.edit?"createSportsman":"updateSportsman"](formSportsman).subscribe

Reply:
[ ejemplo ]